build_compilers/lib/src/modules.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:build/build.dart';
+
+class Module {
+  /// The libraries in the strongly connected cycle.
+  ///
+  /// In most cases without cyclic imports this will contain a single asset.
+  final Set<AssetId> cycle;
+
+  /// The libraries which would have a `.sum` or `.js` output necessary to
+  /// compile this module.
+  ///
+  /// Every import from any of the assets in [cycle] will be covered by the
+  /// modules rooted within this set.
+  final Set<AssetId> dependencies;
```

Comment:
nit: As far as I can tell these are only direct dependencies, I would rename it as such. The comment is also misleading because it claims these are all the required inputs but they aren't.

Reply:
Renamed as `directDependencies` and and updated the comment.